packages/eas-cli/src/credentials/ios/appstore/provisioningProfileAdhoc.ts

```diff
@@ -1,9 +1,219 @@
+      alreadyAdded.push(device);
+    }
+  }
+
+  return alreadyAdded;
+}
+
+async function findDistCert(serial_number: string, isEnterprise: boolean) {
+  const certs = await Certificate.getAsync({
+    query: {
+      filter: {
+        certificateType: CertificateType.IOS_DISTRIBUTION,
+      },
+    },
+  });
+
+  if (serial_number === '__last__') {
+    return certs[certs.length - 1];
+  }
+
+  return certs.find(c => c.attributes.serialNumber === serial_number) ?? null;
+}
+
+async function findProfileByBundleId(bundleId: string, certSerialNumber: string) {
+  let expoProfiles = await getProfilesForBundleId(bundleId);
+  expoProfiles = expoProfiles.filter(
+    profile => profile.attributes.profileType === ProfileType.IOS_APP_INHOUSE
+  );
+
+  expoProfiles = expoProfiles.filter(profile => {
+    return (
+      profile.attributes.name.startsWith('*[expo]') &&
+      profile.attributes.profileState !== ProfileState.EXPIRED
+    );
+  });
+
+  const expoProfilesWithCert: Profile[] = [];
+  // find profiles associated with our development cert
+  for (const profile of expoProfiles) {
+    const certificates = await profile.getCertificatesAsync();
+    if (certificates.some(cert => cert.attributes.serialNumber === certSerialNumber)) {
+      expoProfilesWithCert.push(profile);
+    }
+  }
+
+  if (expoProfilesWithCert) {
+    // there is an expo managed profile with our desired certificate
+    // return the profile that will be valid for the longest duration
+    return {
+      profile: expoProfilesWithCert.sort(sortByExpiration)[expoProfilesWithCert.length - 1],
+      didUpdate: false,
+    };
+  } else if (expoProfiles) {
+    // there is an expo managed profile, but it doesnt have our desired certificate
+    // append the certificate and update the profile
+    const isInHouse = (await Teams.getTeamInfoAsync()).type.toLowerCase() === 'in-house';
+    const distCert = await findDistCert(certSerialNumber, isInHouse);
+    if (!distCert) throw new Error('expected cert not found');
+    const profile = expoProfiles.sort(sortByExpiration)[expoProfiles.length - 1];
+    profile.attributes.certificates = [distCert];
+    return { profile: await profile.regenerateAsync(), didUpdate: true };
+  }
+
+  // there is no valid provisioning profile available
+  return { profile: null, didUpdate: false };
+}
+
+function sortByExpiration(a: Profile, b: Profile): number {
+  return (
+    new Date(a.attributes.expirationDate).getTime() -
+    new Date(b.attributes.expirationDate).getTime()
+  );
+}
+
+async function findProfileById(profileId: string, bundleId: string) {
+  let profiles = await getProfilesForBundleId(bundleId);
+  profiles = profiles.filter(
+    profile => profile.attributes.profileType === ProfileType.IOS_APP_ADHOC
+  );
+  return profiles.find(profile => profile.id === profileId) ?? null;
+}
+
+function uniqueItems<T = any>(items: T[]): T[] {
+  const set = new Set(items);
+  // @ts-ignore: downlevel iteration
+  return [...set];
+}
```

Comment:
You can use `lodash/uniq` instead - https://lodash.com/docs/4.17.15#uniq

Reply:
I think this is fine, `lodash.uniq` is more kilobytes.

Reply:
`lodash` is already installed so why not use it.

Reply:
`lodash` is one of the largest libraries in this project:
```
4.8M	node_modules/lodash
```
I'd try to use as little of it as possible and eventually remove it (we can use individual packages like `lodash.pick` if necessary).

Reply:
So should we use the `lodash.uniq` package then? I'd say no, the `[...new Set(items)]` idiom works well for this purpose and this function @EvanBacon added is 9x smaller than `lodash.uniq`.